BNNApproximation.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from BNN import new_bnn
+import tensorflow as tf
+from bayesianapproximator import BayesianApproximator
+
+class BNNApproximation(BayesianApproximator):
+
+    def __init__(self, state_dimensions, num_acts):
+        super().__init__(state_dimensions, num_acts)
+
+        self.bnn = new_bnn()
+        #TODO refactor this back out into the BNN method
+        self.bnn.compile(optimizer=tf.keras.optimizers.Adam(1e-2),
+            loss=tf.losses.huber_loss)
+        # input = np.array(state_dimensions + self.convert_action(self.num_actions))
+        # self.history = self.bnn.fit(input, 0, batch_size=32,
+        #     epochs=1000, verbose=1)
+
+
+    def update_stats(self, s, a, val=0.0):
+        if self.dimensions.shape != s.shape or not a <= self.num_actions:
```

Comment:
If the actions are 0 indexed this 'a' condition wouldn't work. Should be checking 'a < 0' as well